File: VB/MvpvmNavigation/Services/ModuleTypeResolver.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports MvpvmNavigation.Model

Namespace MvpvmNavigation.Services
    Public Interface IModuleTypesResolver
        Function GetName(ByVal moduleType As ModuleType) As String
        Function GetCaption(ByVal moduleType As ModuleType) As String
        Function GetTypeName(ByVal moduleType As ModuleType) As String
    End Interface
    Friend Class ModuleTypesResolver
        Implements IModuleTypesResolver

        Public Function GetName(ByVal moduleType As ModuleType) As String Implements IModuleTypesResolver.GetName
            If moduleType = MvpvmNavigation.Model.ModuleType.None Then
                Return Nothing
            End If
            Return moduleType.ToString()
        End Function
        Public Function GetCaption(ByVal moduleType As ModuleType) As String Implements IModuleTypesResolver.GetCaption
            If moduleType = MvpvmNavigation.Model.ModuleType.None Then
                Return Nothing
            End If
            Return DevExpress.XtraEditors.EnumDisplayTextHelper.GetDisplayText(moduleType)
        End Function
        Public Function GetTypeName(ByVal moduleType As ModuleType) As String Implements IModuleTypesResolver.GetTypeName
            If moduleType = MvpvmNavigation.Model.ModuleType.None Then
                Return Nothing
            End If
            Select Case moduleType
                Case MvpvmNavigation.Model.ModuleType.ModuleA
                    Return "View1"
                Case MvpvmNavigation.Model.ModuleType.ModuleB
                    Return "View2"
                Case MvpvmNavigation.Model.ModuleType.ModuleA_EditForm
                    Return "View1EditForm"
                Case MvpvmNavigation.Model.ModuleType.ModuleB_EditForm
                    Return "View2EditForm"
            End Select
            Throw New NotSupportedException()
        End Function
    End Class
End Namespace

File: CS/MvpvmNavigation/Services/ModuleTypeResolver.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using MvpvmNavigation.Model;

namespace MvpvmNavigation.Services {
    public interface IModuleTypesResolver {
        string GetName(ModuleType moduleType);
        string GetCaption(ModuleType moduleType);
        string GetTypeName(ModuleType moduleType);
    }
    class ModuleTypesResolver : IModuleTypesResolver {
        public string GetName(ModuleType moduleType) {
            if (moduleType == ModuleType.None)
                return null;
            return moduleType.ToString();
        }
        public string GetCaption(ModuleType moduleType) {
            if (moduleType == ModuleType.None)
                return null;
            return DevExpress.XtraEditors.EnumDisplayTextHelper.GetDisplayText(moduleType);
        }
        public string GetTypeName(ModuleType moduleType) {
            if (moduleType == ModuleType.None)
                return null;
            switch (moduleType) {
                case ModuleType.ModuleA:
                    return "View1";
                case ModuleType.ModuleB:
                    return "View2";
                case ModuleType.ModuleA_EditForm:
                    return "View1EditForm";
                case ModuleType.ModuleB_EditForm:
                    return "View2EditForm";
            }
            throw new NotSupportedException();
        }
    }
}
